src/app/parcels/ActionBar/ActionAndStatusBar.test.tsx

```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import { expect, it } from "@jest/globals";
+import "@testing-library/jest-dom/jest-globals";
+import ActionAndStatusBar, {
+    ActionAndStatusBarProps,
+} from "@/app/parcels/ActionBar/ActionAndStatusBar";
+import { ParcelsTableRow } from "../parcelsTable/types";
+import StyleManager from "@/app/themes";
+import Localization from "@/app/Localization";
+import { SaveParcelStatusResult } from "./Statuses";
+
+jest.mock("@/supabaseClient", () => {
+    return { default: jest.fn() };
+});
+
+const logID = "a2adb0ba-873e-506b-abd1-8cd1782923c8";
+jest.mock("@/logger/logger", () => ({
+    logErrorReturnLogId: jest.fn(() => Promise.resolve(logID)),
+}));
+
+jest.mock("@/app/parcels/ActionBar/Statuses", () => {
+    return jest.fn().mockImplementation(() => (
+        <div>
+            <button data-testid="#status-button">Statuses</button>
+            <div data-testid="#status-menu">
+                <button>Parcel Denied</button>
+                {/* Add more status options as needed */}
+            </div>
+        </div>
+    ));
+});
+
+jest.mock("@/app/parcels/ActionBar/Actions", () => {
+    return jest.fn().mockImplementation(() => (
+        <div>
+            <button data-testid="#action-button">Actions</button>
+            <div data-testid="#action-menu">
+                <button>Download Shopping List</button>
+                {/* Add more status options as needed */}
+            </div>
+        </div>
+    ));
+});
+
+export const MockActionBar: React.FC<ActionAndStatusBarProps> = ({
+    fetchSelectedParcels: fetchSelectedParcels,
+    updateParcelStatuses: onDeleteParcels,
```

Comment:
I dont think you actually have to type these parameters as they should already be typed by the generic type passed into React.FC? Suppose this is purely stylistic so up to you. May be other instances of this pattern elsewhere.

Reply:
They're required here as the code is technically creating a brand new component, but making a new component is unnecessary.

Looking at the other tests I've made, I've implemented the react components differently to this one, so I'll refactor it to make it more like the other ones.